### ISSUE TITLE
Login to Dyson account failed: '403, Forbidden'

### ISSUE THREAD

**Issue by Cad01** [JanJaapKo/DysonPureLink#21]
Hi when i try to install your plugin i get these error in the log

2020-08-10 12:57:07.899 Error: (Dyson) Login to Dyson account failed: '403, Forbidden'
2020-08-10 12:57:07.899 Error: (Dyson) No usable credentials found
2020-08-10 12:57:07.899 Error: (Dyson) 'onStart' failed 'AttributeError':''NoneType' object has no attribute 'device_base_topic''.
2020-08-10 12:57:07.899 Error: (Dyson) ----> Line 415 in '/home/pi/domoticz/plugins/DysonPureLink/plugin.py', function onStart
2020-08-10 12:57:07.899 Error: (Dyson) ----> Line 233 in '/home/pi/domoticz/plugins/DysonPureLink/plugin.py', function onStart

I dubble checked the credentials.

It is a new pure cool not a link. But I think because it not even logs in its not a device error

**Comment by K0enH** [comment]
Here experiencing exactly the same issue

after searching I found the issue to be in dyson.py file.

adding http headers like ` headers={'User-Agent': 'Mozilla/5.0'} )` fixed the issue.
this I was I did on line 37  uri, request_body, verify=False, headers={'User-Agent': 'Mozilla/5.0'} )
and line 54 DYSON_API_URL), verify=False, auth=self._auth, headers={'User-Agent': 'Mozilla/5.0'})

after changing this I was able to succesfully connect

**Comment by Brisket604** [comment]
Hi,
Same issue here, adding header solved thje issue. Thanks @K0enH !!

**Comment by ricorico94** [comment]
Hi,
I faced similar issue with same error message and solved that "403" message by adding the headers: thanks for that.
However, I keep having the errors:

2020-08-25 16:17:26.678 Error: (DysonPurHotCoolCrypto) No usable credentials found
2020-08-25 16:17:26.678 Error: (DysonPurHotCoolCrypto) 'onStart' failed 'AttributeError':''NoneType' object has no attribute 'device_base_topic''.
2020-08-25 16:17:26.678 Error: (DysonPurHotCoolCrypto) ----> Line 415 in '/home/pi/domoticz/plugins/DysonPureLink/plugin.py', function onStart
2020-08-25 16:17:26.678 Error: (DysonPurHotCoolCrypto) ----> Line 233 in '/home/pi/domoticz/plugins/DysonPureLink/plugin.py', function onStart

May I ask you which kind of dyson product you have ? Mine is a Pure Hot+Cool Cryptomic , and the request does not seem sending back any device...

**Comment by K0enH** [comment]
are you using the dyson local way to login or using a cloud account?
in my case I filled in the IP adress of my Dyson, my cloud account and password and the Machine Name from my cloud account.
the specific device type we are using is a "Dyson Pure Cool Link"

hope this helps.

**Comment by ricorico94** [comment]
Hi,
I will try your way, because I was only filling the cloud account name and password. Could you clarify what you mean by "Machine name" here ? Is it the serial number like E3E-EU-XXXX found on the www.dyson.fr website ? Where do you connect for cloud account ?
Because when calling the API, it seems nothing is returned (
2020-08-25 16:17:26.058 (DysonPurHotCoolCrypto) Request URL: 'https://appapi.cp.dyson.com/v1/userregistration/authenticate?country=FR'
2020-08-25 16:17:26.493 (DysonPurHotCoolCrypto) Login OK, JSON response: '{'Account': 'xxxxxxxxxxxxxxxxx', 'Password': 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'}'
2020-08-25 16:17:26.493 (DysonPurHotCoolCrypto) Fetching devices from Dyson Web Services.
2020-08-25 16:17:26.678 (DysonPurHotCoolCrypto) Reply from Dyson: []'
2020-08-25 16:17:26.678 (DysonPurHotCoolCrypto) No devices found in Dyson cloud account

**Comment by K0enH** [comment]
for the device name, user the mobile app on your phone, go to the settings of the device and on top you have the serial number, and below the name of the room.
use the name of the room as device name. ( I am using an iphone so hope you experience the same with the order of the settings on the mobile app)

**Comment by ricorico94** [comment]
Hi,
Thanks K0enH for your clarification. I followed your advice and tried setting up that way, but still same empty answer from the cloud.
I called the Dyson support -hey told me they had not received the 2 emails I sent to them in June.. I resent and they opened a ticket this time, let's see.
For everyone notice, their support guy told me that the proper address is not help@dyson.fr as indicated on their French website, but owner.support@support.dyson.fr . He also told me that it's possible that their system rejects unsolicited emails, meaning that to contact them by email, it might be required to first contact by email.. Unlike what's mentioned on their website as well. A bit unclear...

**Comment by JanJaapKo** [comment]
Hi all,

For some strange reason I did not get a notification from GIT about the creation of thi issue. However, many thanks to @K0enH for providing the solution. I've merged it into the code.

Btw: for questions on setting up the plugin, I wrote some wiki pages: https://github.com/JanJaapKo/DysonPureLink/wiki

The device name is only used to filter on should you have more than one device. It will not help you if there are no devices returned, see topic #20 for more discussion on that.
